pytorch_lightning/plugins/training_type/sharded.py

```diff
@@ -28,10 +31,23 @@
 
 class DDPShardedPlugin(DDPPlugin):
 
+    def __init__(
+        self,
+        parallel_devices: Optional[List[torch.device]] = None,
+        num_nodes: int = 1,
+        cluster_environment: ClusterEnvironment = None,
+        sync_batchnorm: bool = False,
+        is_fp16: bool = False,
+    ):
+        super().__init__(parallel_devices, num_nodes, cluster_environment, sync_batchnorm)
+        self.is_fp16 = is_fp16
+
     def configure_ddp(self):
         self._wrap_optimizers()
         self._model = ShardedDataParallel(
-            LightningShardedDataParallel(self.model), sharded_optimizer=self.lightning_module.trainer.optimizers
+            LightningShardedDataParallel(self.model),
+            sharded_optimizer=self.lightning_module.trainer.optimizers,
+            reduce_buffer_size=2**23 if self.num_nodes > 1 else 0,
```

Comment:
minor comments:
- make 2**23 a private constant for the class
- this should have comments so readers know this is done as a performance optimization